### PR TITLE
Fix for versionCode conflicts when uploading to Play Store

### PR DESCRIPTION
**Pull request by jakubgs**
### Summary:
This fixes how we manage build numbering. So far we simply used count of commits but that doesn't work when we cut custom release branches which vary in length. This can cause conflicts when uploading to Play Store, like in the following build:
https://jenkins.status.im/job/status-react/job/upload_release/21/
This results in error:
```
Google Api Error: apkUpgradeVersionConflict: APK specifies a version code that has already been used.
```
This is specifically caused by the `versionCode` app metadata set in `build.gradle`.

### Review notes (optional):
Currently the script tags the current commit with a new `build-${number}` tag but does not push.
This is left to the user/automation, which right now is done after builds for both platforms succeed.
Maybe a more correct approach would be to push right away, or only after we succeed in uploading, but I'm not sure.

### Testing notes (optional):
Try running it. It prints messages about what's happening to `stderr`.

### Steps to test:
Run `build_no.sh` with or without `--tag` and see what happens.

status: wip

<blockquote></blockquote>

### CODE REVIEW

File: Jenkinsfile.upload_release [hunk 1]
```diff
@@ -75,6 +75,10 @@ node ('macos1'){
       }
     }
 
+    stage('Push build tag') {
+      sh ('git push --tags')
```
Comment (rasom) [inline Jenkinsfile.upload_release]: I doubt that we want to push all intermediate tags to repo, we don't really need to have them on GH.

Comment (jakubgs) [reply Jenkinsfile.upload_release]: Makes sense, I'll push it after the release to Apple & Play store.

**Review by rasom — APPROVED** [review]

**Comment by rasom** [comment]
apk uploaded to https://i.diawi.com/bX8e76 for branch PR-4131

<blockquote><div><strong><a href="https://i.diawi.com/bX8e76">Install - Diawi - Development and In-house Apps Wireless Installation</a></strong></div></blockquote>

**Comment by rasom** [comment]
ipa uploaded to https://i.diawi.com/ib6ENM for branch PR-4131

<blockquote><div><strong><a href="https://i.diawi.com/ib6ENM">Install - Diawi - Development and In-house Apps Wireless Installation</a></strong></div></blockquote>

**Comment by rasom** [comment]
apk uploaded to https://i.diawi.com/AM6nGX for branch PR-4131

<blockquote><div><strong><a href="https://i.diawi.com/AM6nGX">Install - Diawi - Development and In-house Apps Wireless Installation</a></strong></div></blockquote>

**Comment by rasom** [comment]
ipa uploaded to https://i.diawi.com/QrqrW9 for branch PR-4131

<blockquote><div><strong><a href="https://i.diawi.com/QrqrW9">Install - Diawi - Development and In-house Apps Wireless Installation</a></strong></div></blockquote>

**Comment by jakubgs** [comment]
I had to create another branch called `fix/build-numbering-test` which removes the `if` which check if branch names starts with "release/" in order to run this test:

https://jenkins.status.im/job/tests/job/upload_release_test/2/

If you search for `CFBundleVersion` you can see that the build number is `2425` because currently the highest build number tag is `build-2424`.
It fails because the branch name isn't `release/${numbers}`, and that makes app metadata wrong, but I'm not sure if making a branch that looks like a release and pushing it is a good idea.

<blockquote></blockquote>

**Comment by jakubgs** [comment]
The thing that still needs verifying is pushing new tags. In order to do this we need jenkins credentials for git, which I'm checking how to do right now.

**Comment by jakubgs** [comment]
I've fixed pushing tags by explicitly using credentials via `withCredentials` in the Jenkinsfile.
Here's a successful tagging and pushing:
https://jenkins.status.im/job/tests/job/upload_release_test/5/console

<blockquote></blockquote>

**Comment by rasom** [comment]
apk uploaded to https://i.diawi.com/6q3vNe for branch PR-4131

<blockquote><div><strong><a href="https://i.diawi.com/6q3vNe">Install - Diawi - Development and In-house Apps Wireless Installation</a></strong></div></blockquote>

**Comment by rasom** [comment]
ipa uploaded to https://i.diawi.com/jQ5drR for branch PR-4131

<blockquote><div><strong><a href="https://i.diawi.com/jQ5drR">Install - Diawi - Development and In-house Apps Wireless Installation</a></strong></div></blockquote>

**Comment by jakubgs** [comment]
__WARNING:__ Before merging we will have to tag the right commit with the newest build number so we are sure to increment properly.

**Comment by rasom** [comment]
apk uploaded to https://i.diawi.com/WtHgDH for branch PR-4131

<blockquote><div><strong><a href="https://i.diawi.com/WtHgDH">Install - Diawi - Development and In-house Apps Wireless Installation</a></strong></div></blockquote>

**Comment by rasom** [comment]
ipa uploaded to https://i.diawi.com/wjbz7e for branch PR-4131

<blockquote><div><strong><a href="https://i.diawi.com/wjbz7e">Install - Diawi - Development and In-house Apps Wireless Installation</a></strong></div></blockquote>

**Pull request merged**